include/zephyr/net/coap_client.h

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#include <zephyr/net/coap.h>
+
+#define COAP_VERSION 1
+#define COAP_PATH_ELEM_DELIM '/'
+#define COAP_PATH_ELEM_QUERY '?'
+#define MAX_COAP_MSG_LEN CONFIG_COAP_ASYNC_CLIENT_BUFFER_SIZE
+#define COAP_POOL_SLEEP 500
+#define COAP_SEPARATE_TIMEOUT 6000
+#define DEFAULT_RETRY_AMOUNT 5
+
+/**
+ * @typedef coap_client_response_cb_t
+ * @brief Callback for CoAP request.
+ *
+ * This callback is called for responses to CoAP client requests.
+ * It is used to indicate errors, response codes from server or to deliver payload.
+ * Blockwise transfers cause this callback to be called sequentially with increasing payload offset
+ * and only partial content in buffer pointed by payload parameter.
+ *
+ * @param result_code Result code of the response. Negative if there was a failure in send.
+ *                    @ref coap_response_code for positive.
+ * @param offset Payload offset.
+ * @param payload Buffer containing the payload from the response. NULL for empty payload.
+ * @param len Size of the payload.
+ * @param user_data User provided context.
+ * @param last_block Indicates the last block of the response.
+ */
+typedef void (*coap_client_response_cb_t)(uint8_t result_code,
+					  size_t offset, size_t len, const uint8_t *payload,
+					  bool last_block, void *user_data);
+
+/** @cond INTERNAL_HIDDEN */
+struct coap_client_request {
```

Comment:
This structure is used in the public part of the API, it should not be hidden.

Reply:
True, but user should still consider it as opaque. Is there some guideline for the doxygen regarding this kind of case?